### PR TITLE
Updates the JavaScript styleguide with better examples.

### PR DESCRIPTION
**Pull request by poteto**
- Removed redundant examples
- Explicitly show good and bad examples where relevant

This does not add, remove or change any of the rules.

### CODE REVIEW

**Comment by rwjblue** [comment]
:+1:  - Looks good, I'll leave open for a few to see if others have comments....

File: javascript.md [hunk 1]
```diff
@@ -155,46 +160,95 @@ if (foo === 'bar') {
...
+// bad
+const foo = {
+  bar: [1, 2, 3],
+  baz: {
+    a: 'a',
+  },
+}
 ```
 
-+ Skip trailing and leading commas.
```
Comment (danmcclain) [inline javascript.md]: Trailing commas meaning the following, right? 

```js
const boo = [
  a,
  b,
  c, // Trailing <<
];

Comment (poteto) [inline javascript.md]: Yup, the trailing comma rule is [here](https://github.com/dockyard/styleguides/pull/34/files#diff-193a16fbbc467581f0cd8e061c629262L155).

Comment (danmcclain) [inline javascript.md]: D'oh, thanks!

**Pull request merged**